File: scripts/quality/fixtures/invalid.ts
import type { Role } from 'src/entities/role';

export function invalid(first: boolean, second: boolean, third: boolean, fourth: Role): boolean {
  if (first) {
    if (second) {
      if (third) {
        if (fourth) {
          return true;
        }
      }
    }
  }
  return false;
}

File: tools/compliance/tests/fixtures/invalid.rs
pub fn invalid(first: bool, second: bool, third: bool, fourth: bool) -> Option<u8> {
    let mut value = Some(1);
    if first {
        if second {
            if third {
                if fourth {
                    value = None;
                }
            }
        }
    }
    value.unwrap();
    value
}
